aws_analysis_tools/cli/instances_2.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+#
+# © 2016 Krux Digital, Inc.
+#
+
+#
+# Standard libraries
+#
+
+from __future__ import absolute_import
+
+#
+# Third party libraries
+#
+
+from texttable import Texttable
+
+#
+# Internal libraries
+#
+
+import krux_boto
+import krux_ec2.cli
+from krux_ec2.filter import Filter
+
+NAME = 'instances'
+
+
+class Application(krux_ec2.cli.Application):
+
+    # A dict with key=CLI options and value=AWS filters
+    _CLI_TO_AWS = {
+            'group': 'group-name',
+            'name': 'tag:Name',
+            'type': 'instance-type',
+            'zone': 'availability-zone',
+            'state': 'instance-state-name'
+        }
+
+    # List of all options
+    _OPTS = ['group', 'name', 'type', 'zone', 'state']
+
+    # A dict with key=CLI options and value=instance attribute
+    _INSTANCE_ATTR = {
+            'group': lambda i: i.groups[0].name,
```

Comment:
Worth noting that in a VPC context, this will use the group listed first, and ignore any others, which could trip up the user.